tests/integration/test_database.py

```diff
@@ -53,3 +53,37 @@ def write() -> None:
 
     with get_database_connection(database_file) as conn:
         assert conn.cursor().execute("SELECT COUNT(*) FROM test").fetchone() == (1 + n_writers,)
+
+
+def multiproc_read_write(database_file: Path, n_readers: int) -> list[list[tuple[str]]]:
```

Comment:
side note: EFS seems to be build around NFS, and there's mention of intricacies in NFS file locking from [faq](https://www.sqlite.org/faq.html#q5). I have not looked further yet.

Reply:
Good flag, I'll see if this causes any issues. This PR doesn't fix the underlying slowness problem on Cloud, just mitigates it — still some digging to do to find the culprit for the slowdown.